interactive/dashboard/api.py

```diff
@@ -0,0 +1,1036 @@
+class ConversionCalculator(views.APIView):
+    def get(self, request):
+        conversion_request = request.GET.dict()
+        args = [x for x in conversion_request if 'args' in x]
+        arg_dict = {}
+        for key in conversion_request:
+            if 'title' in key:
+                arg_dict.update({conversion_request[key]: float(
+                    conversion_request[key.replace('title', 'value')])})
+                arg_dict.update(
+                    {conversion_request[key] + ' units': (
+                        conversion_request[key.replace('title', 'unit')])})
+        converter = create_unit_converter(
+            conversion_request['initialUnit'],
+            conversion_request['finalUnit'])
+        if arg_dict:
+            new_value = converter(
+                float(conversion_request['initialValue']), arg_dict)
+        else:
+            new_value = converter(float(conversion_request['initialValue']))
+
+        return HttpResponse(new_value)
+
+
+class MusicaReactionsList(views.APIView):
+    def get(self, request):
+        log("****** GET request received MUSICA REACTIONS LIST ******")
+        if not request.session.session_key:
+            request.session.create()
+
+        log("* fetching species list for session id: " +
+              request.session.session_key)
+        reactions = {}
+        path = os.path.join(settings.BASE_DIR, 'configs/' 
+                            + request.session.session_key)
+        if not os.path.isdir(path):
+            log("* detected no data from this user")
+            return JsonResponse({})
+        else:
+            reactions = {"reactions": reaction_musica_names(os.path.join(
+                settings.BASE_DIR, 'configs/'+request.session.session_key)
+                +"/camp_data/reactions.json")}
+            log("returning reactions [mrl]:", reactions)
+            return JsonResponse(reactions)
+
+
+class EvolvingConditions(views.APIView):
+    def get(self, request):
+        log("****** GET request received EVOLVING CONDITIONS ******")
+        if not request.session.session_key:
+            request.session.create()
+        config_path = os.path.join(
+            settings.BASE_DIR, 'configs/'+request.session.session_key
+            +"/my_config.json")
+        log("* config path: "+config_path)
+        with open(config_path) as f:
+            config = json.loads(f.read())
+
+        e = config['evolving conditions']
+        evolving_conditions_list = e.keys()
+
+        # contains a dictionary w/ key as filename and value as header of file
+        file_header_dict = {}
+        for i in evolving_conditions_list:
+            if '.csv' in i or '.txt' in i:
+                log("adding file: "+i)
+                path = os.path.join(os.path.join(
+                    settings.BASE_DIR, 'configs/'
+                    +request.session.session_key), i)
+                with open(path, 'r') as read_obj:
+                    csv_reader = reader(read_obj)
```

Comment:
I'm asking because I don't see an import or declaration for `reader` in this file. We should really think about removing all of the `from _ import *` imports. That is what hiding that this is being imported form what looks like `save.py`.